app/models/pseud.rb

```diff
@@ -418,7 +418,10 @@ def delete_icon
   alias_method :delete_icon?, :delete_icon
 
   def clear_icon
-    self.icon = nil if delete_icon? && !icon.dirty?
+    return unless delete_icon?
+    
+    self.icon = nil if delete_icon?
```

Comment:
🤦 looks like I carried over the wrong condition in my last review, sorry about that!
```suggestion
    self.icon = nil unless icon.dirty?
```